### DISCUSSION TITLE
How to create hyperledger fabric singleton gateway object using node js sdk?

### DISCUSSION THREAD

**Discussion by Argha01** [hyperledger/fabric-sdk-node#648] [Q&A]
We have a existing client application is made of fabric node js sdk to interact with blockchain network. It creates the gateway object and disconnect for every transaction. Can we reuse the gateway object somehow so that burden of creating and destroying gateway objet can be avoided.

Sdk version used : 2.2

I have tried to make it singleton by creating the gateway object at the time of application start and make available throughout the application lifetime but sometime it's giving error like "timed out for discovery".

**Comment by bestbeforetoday** [comment]
The StackOverflow answer to your question might help:

https://stackoverflow.com/a/75007967/3680198
